modelbaker/dbconnector/gpkg_connector.py

```diff
@@ -636,7 +637,31 @@ def get_bags_of_info(self):
                             LEFT JOIN T_ILI2DB_META_ATTRS as meta_attrs_cardinality_max
                             ON LOWER(meta_attrs_cardinality_max.ilielement) = LOWER(cname.iliname||'.'||cprop.columnname) AND meta_attrs_cardinality_max.attr_name = 'ili2db.ili.attrCardinalityMax'
                             WHERE cprop.tag = 'ch.ehi.ili2db.foreignKey' AND meta_attrs_array.attr_value = 'ARRAY'
```

Comment:
What I don't understand is, why it was not sufficient just to return the `meta_attrs_array.attr_value as mapping_type` and then check it in the `suppress_catalogue_reference_layers`? Why is the union select needed?

Reply:
Fair question, taking e.g., `KbS_LV95_V1_3` model from tests:

+ Just returning `meta_attrs_array.attr_value as mapping_type` and commenting `mapping=ARRAY` in `WHERE`:
   + 19 rows returned. I think the other relations are also included.
   + Only the rows with `mapping=ARRAY` have proper cardinality, the rest has `NULL` values for cardinalities.

+ Using the `UNION ALL` :
    + 4 rows returned. One per BAG OF in the model. 
    + Each row includes proper cardinality.

Nonetheless, I think the UNION ALL query looks a bit harder to maintain. I'm reverting to keep it simple.